File: day5/index1.js
const { readLines, parseInt } = require("../lib");

const jumps = readLines("./input").map(parseInt);

const l = jumps.length;
let count = 0;
let i = 0;

while (i >= 0 && i < l) {
  const jump = jumps[i];
  jumps[i]++;
  i += jump;
  count++;
}

console.log(count);

File: clojure/day05/index1.clj
(require '[clojure.string :as s])

(def parse-int #(Integer/parseInt %))

(def jumps (->> "./input"
  slurp
  s/split-lines
  (map parse-int)
  (into [])
))

(def l (count jumps))

(def cnt (loop [
  i 0
  jumps jumps
  cnt 0
] (
  if (or (< i 0) (>= i l)) cnt (
    recur (+ i (get jumps i)) (update jumps i inc) (inc cnt)
  )
)))

(println cnt)